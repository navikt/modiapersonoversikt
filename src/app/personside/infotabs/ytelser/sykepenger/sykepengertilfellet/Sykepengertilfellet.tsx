import DescriptionList, {
    type DescriptionListEntries,
    fjernEntriesUtenVerdi
} from '../../../../../../components/DescriptionList';
import type { Forsikring, Sykepenger, Sykmelding } from '../../../../../../models/ytelse/sykepenger';
import {
    NOKellerNull,
    convertBoolTilJaNei,
    datoEllerNull,
    periodeEllerNull
} from '../../../../../../utils/string-utils';
import YtelserInfoGruppe from '../../felles-styling/YtelserInfoGruppe';

interface Props {
    sykepenger: Sykepenger;
}

function getForsikringEntries(forsikring: Forsikring | null): DescriptionListEntries {
    if (!forsikring) {
        return {};
    }
    return {
        Forsikring: NOKellerNull(forsikring.premiegrunnlag),
        'Gyldig forsikring': convertBoolTilJaNei(forsikring.erGyldig),
        Forsikringsperiode: periodeEllerNull(forsikring.forsikret)
    };
}

function getYrkesskadeEntries(sykeMeldinger: Sykmelding[]): DescriptionListEntries {
    const sisteSykemelding = sykeMeldinger[0];
    if (!sisteSykemelding || !sisteSykemelding.gjelderYrkesskade) {
        return {};
    }
    const yrkesSkade = sisteSykemelding.gjelderYrkesskade;
    return {
        Yrkesskade: yrkesSkade.yrkesskadeart,
        Yrkesskadedato: datoEllerNull(yrkesSkade.skadet),
        Vedtaksdato: datoEllerNull(yrkesSkade.vedtatt)
    };
}

function Sykepengertilfellet({ sykepenger }: Props) {
    const sykepengeTilfelletEntries: DescriptionListEntries = {
        'Sykemeldt fra og med': datoEllerNull(sykepenger.sykmeldtFom),
        'Forbrukte dager': sykepenger.forbrukteDager,
        'Untatt aktivitet': sykepenger.unntakAktivitet,
        'Midlertidig stans': sykepenger.midlertidigStanset,
        ...fjernEntriesUtenVerdi({
            Stans√•rsak: sykepenger.stansaarsak,
            Sanksjonperiode: periodeEllerNull(sykepenger.sanksjon)
        }),
        Ferieperioder: periodeEllerNull(sykepenger.ferie1),
        ...fjernEntriesUtenVerdi({
            Ferieperioder2: periodeEllerNull(sykepenger.ferie2)
        }),
        ...getForsikringEntries(sykepenger.forsikring),
        ...getYrkesskadeEntries(sykepenger.sykmeldinger)
    };

    return (
        <YtelserInfoGruppe tittel="Sykepengetilfellet">
            <DescriptionList entries={sykepengeTilfelletEntries} />
        </YtelserInfoGruppe>
    );
}

export default Sykepengertilfellet;
