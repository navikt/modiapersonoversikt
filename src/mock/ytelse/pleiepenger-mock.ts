import faker from 'faker/locale/nb_NO';
import moment from 'moment';

import navfaker from 'nav-faker/dist/index';
import {
    Arbeidsforhold,
    Periode,
    Pleiepengeperiode,
    Pleiepengerettighet,
    PleiepengerResponse,
    Vedtak
} from '../../models/ytelse/pleiepenger';
import { fyllRandomListe } from '../utils/mock-utils';
import { aremark } from '../person/aremark';
import { moss } from '../person/moss';
import { pleiepengerTestData } from '../../app/personside/infotabs/ytelser/pleiepenger/pleiepengerTestData';
import { backendDatoformat } from '../../utils/dateUtils';

export function getMockPleiepenger(fødselsnummer: string): PleiepengerResponse {
    if (fødselsnummer === aremark.fødselsnummer) {
        return {
            pleiepenger: [
                {
                    ...pleiepengerTestData,
                    barnet: moss.fødselsnummer
                }
            ]
        };
    }

    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'pleiepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {
            pleiepenger: null
        };
    }

    return {
        pleiepenger: fyllRandomListe<Pleiepengerettighet>(() => getMockPleiepengerettighet(fødselsnummer), 2)
    };
}

export function getMockPleiepengerettighet(fødselsnummer: string): Pleiepengerettighet {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'pleiepenger');

    const pleiepengeDager = 1300;
    const forbrukteDager = navfaker.random.integer(pleiepengeDager);
    const restDager = pleiepengeDager - forbrukteDager;
    return {
        barnet: navfaker.personIdentifikator.fødselsnummer(),
        omsorgsperson: fødselsnummer,
        andreOmsorgsperson: navfaker.personIdentifikator.fødselsnummer(),
        restDagerFomIMorgen: restDager,
        forbrukteDagerTomIDag: forbrukteDager,
        pleiepengedager: pleiepengeDager,
        restDagerAnvist: restDager,
        perioder: fyllRandomListe<Pleiepengeperiode>(() => getPleiepengeperiode(), 10)
    };
}

function getPleiepengeperiode(): Pleiepengeperiode {
    return {
        fom: moment(faker.date.past(2)).format(backendDatoformat),
        antallPleiepengedager: navfaker.random.integer(20),
        arbeidsforhold: fyllRandomListe<Arbeidsforhold>(() => getArbeidsforhold(), 2),
        vedtak: fyllRandomListe<Vedtak>(() => getVedtak(), 3)
    };
}

function getArbeidsforhold(): Arbeidsforhold {
    return {
        arbeidsgiverNavn: faker.company.companyName(),
        arbeidsgiverKontonr: Number(faker.finance.account(11)).toString(),
        inntektsperiode: 'Månedssats',
        inntektForPerioden: Math.round(Number(faker.finance.amount(5000, 50000))),
        refusjonTom: moment(faker.date.past(2)).format(backendDatoformat),
        refusjonstype: 'Ikke refusjon',
        arbeidsgiverOrgnr: '1234567890',
        arbeidskategori: 'Arbeidstaker'
    };
}

function getVedtak(): Vedtak {
    return {
        periode: getPeriode(),
        kompensasjonsgrad: navfaker.random.vektetSjanse(0.5) ? 100 : navfaker.random.integer(100),
        utbetalingsgrad: navfaker.random.vektetSjanse(0.5) ? 100 : navfaker.random.integer(100),
        anvistUtbetaling: moment(faker.date.past(2)).format(backendDatoformat),
        bruttobeløp: Number(faker.commerce.price()),
        dagsats: navfaker.random.integer(70),
        pleiepengegrad: navfaker.random.integer(100)
    };
}

function getPeriode(): Periode {
    const fom = moment(faker.date.past(2));
    const tom = moment(fom).add(faker.random.number(40), 'days');
    return {
        fom: fom.format(backendDatoformat),
        tom: tom.format(backendDatoformat)
    };
}
