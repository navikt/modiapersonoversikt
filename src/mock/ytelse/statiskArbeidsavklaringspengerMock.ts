export const statiskArbeidsavklaringspengerMock = {
    barnMedStonad: 2,
    barnetillegg: 1500,
    beregningsgrunnlag: 30000,
    dagsats: 500,
    kildesystem: 'KELVIN',
    periode: {
        fraOgMedDato: '2025-01-01',
        tilOgMedDato: '2025-12-31'
    },
    rettighetsType: 'AAP',
    saksnummer: '2023123456',
    status: 'LÃ˜PENDE',
    vedtakId: 'vedtak-001',
    vedtaksdato: '2023-02-01',
    opphorsAarsak: 'FERDIGBEHANDLET'
};
