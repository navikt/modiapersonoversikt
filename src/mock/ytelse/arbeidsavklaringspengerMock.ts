import { fakerNB_NO as faker } from '@faker-js/faker';
import dayjs from 'dayjs';
import navfaker from 'nav-faker/dist/index';
import { statiskArbeidsavklaringspengerMock } from 'src/mock/ytelse/statiskArbeidsavklaringspengerMock';
import type {
    Arbeidsavklaringspenger,
    ArbeidsavklaringspengerResponse
} from 'src/models/ytelse/arbeidsavklaringspenger';
import { backendDatoformat } from 'src/utils/date-utils';
import { aremark } from '../persondata/aremark';
import { fyllRandomListe } from '../utils/mock-utils';

export function getMockArbeidsavklaringspengerResponse(fnr: string): ArbeidsavklaringspengerResponse {
    if (fnr === aremark.personIdent) {
        return [statiskArbeidsavklaringspengerMock];
    }

    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}arbeidsavklaringspenger`);

    if (navfaker.random.vektetSjanse(0.3)) {
        return [];
    }

    return fyllRandomListe<Arbeidsavklaringspenger>(() => getMockArbeidsavklaringspenger(fnr), 3);
}

function getMockArbeidsavklaringspenger(fnr: string): Arbeidsavklaringspenger {
    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    const fomDato = dayjs(faker.date.past({ years: 2 })).format(backendDatoformat);
    const tomDato = dayjs(fomDato).add(faker.number.int(40), 'days').format(backendDatoformat);

    return {
        barnMedStonad: 2,
        barnetillegg: 1500,
        beregningsgrunnlag: 30000,
        dagsats: 500,
        kildesystem: 'Arena',
        periode: {
            fraOgMedDato: fomDato,
            tilOgMedDato: tomDato
        },
        rettighetsType: 'BISTANDSBEHOV',
        saksnummer: '2023123456',
        status: 'LÃ˜PENDE',
        vedtakId: faker.number.int(100).toString(),
        vedtaksdato: '2023-02-01',
        opphorsAarsak: 'FERDIGBEHANDLET'
    };
}
