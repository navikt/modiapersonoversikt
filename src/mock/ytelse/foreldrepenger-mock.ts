import * as faker from 'faker/locale/nb_NO';
import * as moment from 'moment';

import navfaker from 'nav-faker/dist/index';

import {
    Arbeidsforhold,
    Foreldrepengerperiode,
    ForeldrepengerResponse,
    Foreldrepengerettighet
} from '../../models/ytelse/foreldrepenger';
import { getPeriode } from '../person/periodeMock';
import { backendDatoformat, fyllRandomListe, vektetSjanse } from '../utils/mock-utils';
import { getHistoriskUtbetaling, getKommendeUtbetaling } from './ytelse-utbetalinger-mock';
import { HistoriskUtbetaling, KommendeUtbetaling } from '../../models/ytelse/ytelse-utbetalinger';

export function getMockForeldrepenger(fødselsnummer: string): ForeldrepengerResponse {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'foreldrepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {
            foreldrepenger: null
        };
    }

    return {
        foreldrepenger: getForeldrepengerettighetMock(fødselsnummer)
    };
}

export function getForeldrepengerettighetMock(fødselsnummer: string): Foreldrepengerettighet {
    return {
        forelder: fødselsnummer,
        andreForeldersFnr: navfaker.personIdentifikator.fødselsnummer(),
        antallBarn: navfaker.random.integer(5),
        barnetsFødselsdato: moment(faker.date.recent()).format(backendDatoformat),
        dekningsgrad: navfaker.random.integer(95),
        fedrekvoteTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        mødrekvoteTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        foreldrepengetype: foreldrePengeType(),
        graderingsdager: navfaker.random.integer(100),
        restDager: navfaker.random.integer(50),
        rettighetFom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        eldsteIdDato: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        foreldreAvSammeKjønn: vektetSjanse(faker, 0.5) ? 'Begge er pappaer' : null,
        periode: fyllRandomListe<Foreldrepengerperiode>(() => getForeldrepengerperiodeMock(fødselsnummer), 5),
        slutt: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        arbeidsforhold: fyllRandomListe<Arbeidsforhold>(() => getArbeidsforholdMock(fødselsnummer), 5),
        erArbeidsgiverperiode: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        arbeidskategori: vektetSjanse(faker, 0.5) ? 'Arbeidstaker' : null
    };
}

export function getForeldrepengerperiodeMock(fødselsnummer: string): Foreldrepengerperiode {
    return {
        fødselsnummer: fødselsnummer,
        harAleneomsorgFar: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        harAleneomsorgMor: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        arbeidsprosentMor: vektetSjanse(faker, 0.5) ? navfaker.random.integer(100) : null,
        avslagsårsak: vektetSjanse(faker, 0.5) ? 'Avslag' : null,
        avslått: vektetSjanse(faker, 0.5) ? 'Avslått' : null,
        disponibelGradering: vektetSjanse(faker, 0.5) ? navfaker.random.integer(75) : null,
        erFedrekvote: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        erMødrekvote: vektetSjanse(faker, 0.5) ? faker.random.boolean() : null,
        forskyvelsesårsak1: vektetSjanse(faker, 0.5) ? 'FÅRSAK1' : null,
        forskyvelsesperiode1: vektetSjanse(faker, 0.5) ? getPeriode() : null,
        forskyvelsesårsak2: vektetSjanse(faker, 0.5) ? 'FÅRSAK2' : null,
        forskyvelsesperiode2: vektetSjanse(faker, 0.5) ? getPeriode() : null,
        foreldrepengerFom: moment(faker.date.past(5)).format(backendDatoformat),
        midlertidigStansDato: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        morSituasjon: vektetSjanse(faker, 0.5) ? faker.lorem.words(5) : null,
        rettTilFedrekvote: vektetSjanse(faker, 0.5) ? 'Rett til fedrekvote' : 'Ingen rett til fedrekvote',
        rettTilMødrekvote: vektetSjanse(faker, 0.5) ? 'Rett til mødrekvote' : 'Ingen rett til mødrekvote',
        stansårsak: vektetSjanse(faker, 0.5) ? 'Avsluttet' : null,
        historiskeUtbetalinger: fyllRandomListe<HistoriskUtbetaling>(() => getHistoriskUtbetaling(faker), 5),
        kommendeUtbetalinger: fyllRandomListe<KommendeUtbetaling>(() => getKommendeUtbetaling(faker), 5)
    };
}

function getArbeidsforholdMock(fødselsnummer: string): Arbeidsforhold {
    return {
        navn: faker.company.companyName(),
        kontonr: Number(faker.finance.account(11)).toString(),
        inntektsperiode: vektetSjanse(faker, 0.5) ? 'Månedlig' : 'Årlig',
        inntektForPerioden: Math.round(Number(faker.finance.amount(5000, 50000))),
        sykepengerFom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        refusjonTom: vektetSjanse(faker, 0.5) ? moment(faker.date.recent()).format(backendDatoformat) : null,
        refusjonstype: 'Ikke refusjon'
    };
}

function foreldrePengeType(): string {
    return navfaker.random.arrayElement([
        'Fødselspenger', 'Svangerskapspenger', 'Engangsstønad'
    ]);
}