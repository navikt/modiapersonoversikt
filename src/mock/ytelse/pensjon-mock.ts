import { fakerNB_NO as faker } from '@faker-js/faker';
import dayjs from 'dayjs';
import navfaker from 'nav-faker/dist/index';
import { statiskPensjonMock } from 'src/mock/ytelse/statiskPensjonMock';
import type { Pensjon, PensjonResource } from 'src/models/ytelse/pensjon';
import { backendDatoformat } from 'src/utils/date-utils';
import { aremark } from '../persondata/aremark';
import { fyllRandomListe } from '../utils/mock-utils';

export function getMockPensjon(fnr: string): PensjonResource {
    if (fnr === aremark.personIdent) {
        return [statiskPensjonMock];
    }

    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    if (navfaker.random.vektetSjanse(0.3)) {
        return null;
    }

    return fyllRandomListe<Pensjon>(() => getMockPensjonYtelser(fnr), 3);
}

function getMockPensjonYtelser(fnr: string): Pensjon {
    faker.seed(Number(fnr));
    navfaker.seed(`${fnr}pensjon`);

    const fomDato = dayjs(faker.date.past({ years: 2 })).format(backendDatoformat);
    const tomDato = dayjs(fomDato).add(faker.number.int(40), 'days').format(backendDatoformat);

    return {
        sakid: faker.string.alphanumeric(8),
        fomDato,
        tomDato,
        enhetId: '0129',
        sakType: 'AFP',
        sakStatus: 'OPPRETTET'
    };
}
