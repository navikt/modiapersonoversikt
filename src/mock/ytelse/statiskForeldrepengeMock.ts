import { Fødsel } from '../../models/ytelse/foreldrepenger';

export const statiskKommendeUtbetaling = {
    vedtak: { fra: '2018-01-14', til: '2017-07-15' },
    utbetalingsgrad: 79,
    utbetalingsdato: '2018-06-25',
    bruttobeløp: 466,
    arbeidsgiverNavn: 'Vedvik - Lunde',
    arbeidsgiverOrgNr: '1234567890',
    arbeidsgiverKontonr: '549801574',
    dagsats: 705,
    saksbehandler: 'Mari Johannessen',
    type: 'KONTOØVERFØRING'
};

export const statiskForeldrepengeMock: Fødsel = {
    forelder: '10108000398',
    andreForeldersFnr: '03063850003',
    antallBarn: 5,
    barnetsFødselsdato: '2019-02-11',
    dekningsgrad: 34,
    fedrekvoteTom: '2019-02-10',
    mødrekvoteTom: null,
    foreldrepengetype: 'Fødselspenger',
    graderingsdager: 55,
    restDager: 41,
    rettighetFom: '2017-11-21',
    eldsteIdDato: null,
    foreldreAvSammeKjønn: null,
    periode: [
        {
            fødselsnummer: '10108000398',
            harAleneomsorgFar: null,
            harAleneomsorgMor: null,
            arbeidsprosentMor: 58,
            avslagsårsak: 'Avslag',
            avslått: 'Avslått',
            disponibelGradering: 9,
            erFedrekvote: null,
            erMødrekvote: null,
            forskyvelsesårsak1: null,
            forskyvelsesperiode1: { fra: '2014-10-08T14:22:55.629Z', til: '2014-08-22T10:50:40.296Z' },
            forskyvelsesårsak2: null,
            forskyvelsesperiode2: { fra: '2014-12-09T16:29:21.344Z', til: '2014-08-26T10:23:03.605Z' },
            foreldrepengerFom: '2017-04-29',
            midlertidigStansDato: '2019-02-10',
            morSituasjon: 'nobis sit quo et quis',
            rettTilFedrekvote: 'Rett til fedrekvote',
            rettTilMødrekvote: 'Ingen rett til mødrekvote',
            stansårsak: 'Avsluttet',
            kommendeUtbetalinger: [statiskKommendeUtbetaling]
        }
    ],
    slutt: null,
    arbeidsforhold: [
        {
            arbeidsgiverNavn: 'Bakke - Solheim',
            arbeidsgiverKontonr: '35179178779',
            inntektsperiode: 'Månedlig',
            inntektForPerioden: 40062,
            sykepengerFom: '2019-02-10',
            refusjonTom: '2019-02-11',
            refusjonstype: 'Ikke refusjon'
        },
        {
            arbeidsgiverNavn: 'Bakke - Solheim',
            arbeidsgiverKontonr: '35179178779',
            inntektsperiode: 'Månedlig',
            inntektForPerioden: 40062,
            sykepengerFom: '2019-02-10',
            refusjonTom: '2019-02-11',
            refusjonstype: 'Ikke refusjon'
        }
    ],
    erArbeidsgiverperiode: null,
    arbeidskategori: 'Arbeidstaker',
    termin: '2019-02-11'
};
