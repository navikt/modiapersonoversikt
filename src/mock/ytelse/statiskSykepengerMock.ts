import type { Sykepenger } from '../../models/ytelse/sykepenger';
import { statiskKommendeUtbetaling } from './statiskForeldrepengeMock';

const arbeidsforhold = {
    arbeidsgiverNavn: 'Bærlevåg Burger og Bitcoin',
    arbeidsgiverKontonr: '87865275970',
    inntektsperiode: 'Månedssats',
    inntektForPerioden: 39691,
    refusjonTom: '2018-03-12',
    refusjonstype: 'Ikke refusjon',
    sykepengerFom: '2018-02-18'
};

const statiskUtbetalingPåVentMock = {
    vedtak: { fra: '2017-10-13', til: '2018-10-10' },
    utbetalingsgrad: 62,
    oppgjørstype: 'Spesidaler i pung',
    arbeidskategori: 'Omreisende skald',
    stansårsak: 'Pga mistenkelig oppførsel',
    ferie1: { fra: '2017-07-04', til: '2018-10-29' },
    ferie2: { fra: '2017-07-13', til: '2019-02-03' },
    sanksjon: { fra: '2018-04-24', til: '2018-10-31' },
    sykmeldt: { fra: '2017-11-20', til: '2018-06-24' }
};

const sykmelding = {
    sykmelder: 'Lommelegen AS',
    behandlet: '2018-04-01',
    sykmeldt: { fra: '2014-02-09T19:24:27.357Z', til: '2014-05-11T02:18:25.709Z' },
    sykmeldingsgrad: 97,
    gjelderYrkesskade: null,
    gradAvSykmeldingListe: [
        {
            gradert: { fra: '2014-07-28T10:47:27.559Z', til: '2014-01-03T06:49:39.367Z' },
            sykmeldingsgrad: 89
        }
    ]
};

export const statiskSykepengerMock: Sykepenger = {
    fødselsnummer: '10108000398',
    sykmeldtFom: '2019-02-06',
    forbrukteDager: 23,
    ferie1: { fra: '2014-01-19T06:56:39.720Z', til: '2014-01-14T00:16:11.870Z' },
    ferie2: null,
    sanksjon: { fra: '2014-01-19T01:31:21.521Z', til: '2014-09-05T11:21:25.288Z' },
    stansårsak: null,
    unntakAktivitet: null,
    forsikring: {
        forsikringsordning: 'Betaler beskyttelsespenger til Mafia',
        premiegrunnlag: 837,
        erGyldig: false,
        forsikret: null
    },
    sykmeldinger: [sykmelding, sykmelding],
    kommendeUtbetalinger: [statiskKommendeUtbetaling, statiskKommendeUtbetaling],
    utbetalingerPåVent: [statiskUtbetalingPåVentMock, statiskUtbetalingPåVentMock],
    bruker: '10108000398',
    midlertidigStanset: null,
    slutt: '2019-01-16',
    arbeidsforholdListe: [arbeidsforhold, arbeidsforhold],
    erArbeidsgiverperiode: true,
    arbeidskategori: 'Ærlig arbeid'
};
