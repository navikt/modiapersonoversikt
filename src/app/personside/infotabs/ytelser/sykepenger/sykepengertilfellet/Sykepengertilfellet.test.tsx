import * as renderer from 'react-test-renderer';
import { statiskSykepengerMock } from '../../../../../../mock/ytelse/statiskSykepengerMock';
import Sykepengertilfellet from './Sykepengertilfellet';

test('Sykepengertilfellet matcher snapshot', () => {
    const resultat = renderer.create(<Sykepengertilfellet sykepenger={statiskSykepengerMock} />);

    expect(resultat.toJSON()).toMatchSnapshot();
});
