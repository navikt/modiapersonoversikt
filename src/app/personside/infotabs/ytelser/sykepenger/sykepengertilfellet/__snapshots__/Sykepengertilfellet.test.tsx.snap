// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sykepengertilfellet matcher snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 dt {
  color: #645f5a;
}

.c1 > div {
  margin: 0.5625rem 0;
  padding-right: 0.7rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c0 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c0 > h3 {
  margin-bottom: 1rem;
}

<article
  aria-describedby="Helt tilfeldig ID"
  className="c0"
>
  <h3
    className="typo-ingress"
    id="Helt tilfeldig ID"
  >
    Sykepengetilfellet
  </h3>
  <dl
    className="c1"
  >
    <div>
      <dt
        className="typo-normal"
      >
        Sykemeldt fra og med
      </dt>
      <dd
        className="typo-element"
      >
        06.02.2019
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Forbrukte dager
      </dt>
      <dd
        className="typo-element"
      >
        23
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Maksdato
      </dt>
      <dd
        className="typo-element"
      >
        16.01.2019
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Untatt aktivitet
      </dt>
      <dd
        className="typo-element"
      >
        —
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Midlertidig stans
      </dt>
      <dd
        className="typo-element"
      >
        —
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Sanksjonperiode
      </dt>
      <dd
        className="typo-element"
      >
        19.01.2014 - 05.09.2014
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Ferieperioder
      </dt>
      <dd
        className="typo-element"
      >
        19.01.2014 - 14.01.2014
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Forsikring
      </dt>
      <dd
        className="typo-element"
      >
        837.00 NOK
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Gyldig forsikring
      </dt>
      <dd
        className="typo-element"
      >
        Nei
      </dd>
    </div>
    <div>
      <dt
        className="typo-normal"
      >
        Forsikringsperiode
      </dt>
      <dd
        className="typo-element"
      >
        —
      </dd>
    </div>
  </dl>
</article>
`;
